torchgeo/datasets/oscd.py

```diff
@@ -0,0 +1,308 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""OSCD dataset."""
+
+import glob
+import os
+from typing import Callable, Dict, List, Optional, Sequence, Union
+
+import matplotlib.pyplot as plt
+import numpy as np
+import rasterio
+import torch
+from matplotlib.figure import Figure
+from PIL import Image
+from torch import Tensor
+
+from ..datasets.utils import draw_semantic_segmentation_masks
+from .geo import VisionDataset
+from .utils import download_url, extract_archive, sort_sentinel2_bands
+
+
+class OSCD(VisionDataset):
+    """OSCD dataset.
+
+    The Onera Satellite Change Detection dataset addresses the issue of detecting
+    changes between satellite images from different dates.
+
+    Dataset features:
+
+    * 24 pairs of images from all over the world between 2015-2018
+    * binary change masks representing urban changes such as new buildings or roads
+    * 13 spectral bands
+    * Images vary in spatial resolution between 10m, 20m, and 60m
+
+    Dataset format:
+
+    * images are 13-channel tifs
+    * masks are single-channel pngs where no change = 0, change = 255
+
+    Dataset classes:
+
+    1. no change
+    2. change
+
+    If you use this dataset in your research, please cite the following paper:
+
+    * https://arxiv.org/abs/1810.08468
+    """
+
+    url = "https://drive.google.com/file/d/1jidN0DKEIybOrP0j7Bos8bGDDq3Varj3"
```

Comment:
This actually points to my own self hosted version of the dataset. It seems though that the authors now host the dataset outside of IEEE DataPort. You can find the links [here](https://rcdaudt.github.io/oscd/). We should modify this to use those urls for download instead.